go/cmd/dolt/commands/tag.go

```diff
@@ -76,6 +86,14 @@ func (cmd TagCmd) Exec(ctx context.Context, commandStr string, args []string, dE
 	help, usage := cli.HelpAndUsagePrinters(cli.CommandDocsForCommandString(commandStr, tagDocs, ap))
 	apr := cli.ParseArgsOrDie(ap, args, help)
 
+	queryist, sqlCtx, closeFunc, err := cliCtx.QueryEngine(ctx)
+	if err != nil {
+		return handleStatusVErr(err)
+	}
+	if closeFunc != nil {
+		defer closeFunc()
+	}
+
 	// list tags
 	if len(apr.Args) == 0 {
 		var verr errhand.VerboseError
```

Comment:
There is no reason to have this line three times - lines 99, 115, 133. Just define once at the top of the function

Reply:
True, though it is nice to have self-contained sections of the code and not have to worry about fall-through. For example, lines 97-111 deal with listing tags, and that's all, no need to look elsewhere.
That being said, most of these blocks are just doing some param validation and then delegate to a small function, so I'm going to move the logic into those functions. Should make things a bit cleaner.